### PR TITLE
#836  Test categories matrix sorting

### LINKED ISSUE #836: catalog.py:27-30: Create tests for the catalog matrix....
The puzzle `822-663573d7` from #822 has to be resolved: 

https://github.com/fidals/shopelectro/blob/c89b807552a45099cad752b4121f40d8836eb024/shopelectro/views/catalog.py#L27-L30

The puzzle was created by duker33 on 30-Apr-19. 

  

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by duker33**
Closes #836 

Didn't test block items limiting. The test should be too complex, for a such minor limiting feature

### CODE REVIEW

File: shopelectro/views/catalog.py [hunk 1]
```diff
@@ -18,16 +18,11 @@
...
-
 # block numeric indexes to limit
 MATRIX_BLOCKS_TO_LIMIT = [3, 5]
 MATRIX_BLOCK_SIZE = 7
 
 
-# @todo #822:60m  Create tests for the catalog matrix.
-#  This cases are would be good to test:
-#  - Matrix sorting by page position
-#  - Block items limiting
```
Comment (artemiy312) [inline shopelectro/views/catalog.py]: > Block items limiting

It seems you still haven't test this

**Comment by duker33** [comment]
> It seems you took changes from different branch

sorry, i should wait for rebase on #850 task's code merging

File: shopelectro/views/catalog.py (hunk 1 above)
Comment (duker33) [reply shopelectro/views/catalog.py]: y, did it intentionally. Updated PR body about it

**Comment by duker33** [comment]
@artemiy312 , it's ready for review

File: shopelectro/tests/tests_views.py [hunk 2]
```diff
@@ -543,6 +543,45 @@ def test_category_matrix_page(self):
             self.assertEqual(from_db.name, from_app.a.text)
 
 
+@tag('fast', 'catalog')
+class CategoriesMatrix(BaseCatalogTestCase):
+
+    fixtures = ['dump.json']
+    URL = reverse_lazy('custom_page', kwargs={'page': 'catalog'})
```
Comment (artemiy312) [inline shopelectro/tests/tests_views.py]: You  can move this to `get_page` and use `reverse` instead of `reverse_lazy`

**Pull request merged**